lib/database/sthDatabase.js

```diff
@@ -313,6 +313,60 @@ function getCollection(params, options, callback) {
         return process.nextTick(callback.bind(null, error));
     }
 
+    //Call fetchCollection after creating database connection if connection was not restored after database restarted
+    if (isConnectionAlive()) {
+        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+    } else {
+        reconnectDB(callback, function() {
+            fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+        });
+    }
+}
+
+/**
+ * Check whether database is connected or not
+ */
+function isConnectionAlive() {
+    let dbConnected = true;
+    if (db === null || !db.serverConfig.isConnected()) {
```

Comment:
Since 3.1 driver mongo client has isConnected
https://mongodb.github.io/node-mongodb-native/3.1/api/MongoClient.html#isConnected
So you can check client.isConnect()

Reply:
Done in commit https://github.com/telefonicaid/fiware-sth-comet/pull/574/commits/28233600ab65dbad33f0ebac1b4426c2af834677. Thanks :)